### PR TITLE
Drop support for Python versions < 3.6

### PR DESCRIPTION
**Pull request by sondreso**
Drop support for Python versions below 3.6.

**Approach**
Remove code targeting python 2.7. In addition this PR also switches to Github actions for testing, so it will require admin merge and switching required tests. Adds testing for Python 3.7 and 3.8 on Ubuntu.

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,9 +1,8 @@
-black==19.10b0 ; python_version >= '3.6'
+black==19.10b0
 pytest<6
 pytest-qt
-mock ; python_version >= '3'
-mock<4; python_version < '3'
-schemathesis==1.3.4 ; python_version >= '3'
+mock
```
Comment (dotfloat) [inline dev-requirements.txt:4]: Is this required when we're using `unittest.mock`?

File: ert_gui/__init__.py [hunk 2]
```diff
@@ -1,10 +1,7 @@
 import sys
 
-REQUIRED_VERSION_HEX = 0x02070000
-
-
-if sys.hexversion < REQUIRED_VERSION_HEX:
-    raise Exception("ERT GUI Python requires at least version 2.7 of Python")
+if sys.version_info < (3,6):
```
Comment (dotfloat) [inline ert_gui/__init__.py:3]: Is this necessary? I don't know of any other project that checks the Python version to tell you to sod off.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      env:
+        LD_LIBRARY_PATH: "${{ env.INSTALL_DIR }}/lib:${{ env.INSTALL_DIR }}/lib64"
+        DYLD_LIBRARY_PATH: "${{ env.INSTALL_DIR }}/lib:${{ env.INSTALL_DIR }}/lib64"
+        PYTHONPATH: ${{ env.INSTALL_DIR }}/lib/python${{ matrix.python-version }}/site-packages:${{ env.INSTALL_DIR }}/lib/python${{ matrix.python-version }}/dist-packages
+      run: |
```
Comment (dotfloat) [inline .github/workflows/build.yml:32]: Replace bash `if` with:

```yaml
- name: Install Ubuntu Dependencies
  if: ${{ matrix.os == "ubuntu-latest" }}
  run: |
    sudo apt-get install -y ...
```

I'm not entirely certain about the syntax, but let's avoid doing bash `if`s.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      env:
+        LD_LIBRARY_PATH: "${{ env.INSTALL_DIR }}/lib:${{ env.INSTALL_DIR }}/lib64"
```
Comment (dotfloat) [inline .github/workflows/build.yml:29]: Can we try `sudo make install` instead of dealing with this path garbage?

**Comment by dotfloat** [comment]
LGTM so far on my first look-through. I'll give it another look later today.

File: dev-requirements.txt (hunk 1 above)
Comment (jondequinor) [reply dev-requirements.txt:4]: It's in the py3 std lib.

Comment (sondreso) [reply dev-requirements.txt:4]: Forgot to remove this one!

File: ert_gui/__init__.py (hunk 2 above)
Comment (sondreso) [reply ert_gui/__init__.py:3]: Agree, will remove it

File: .github/workflows/build.yml (hunk 4 above)
Comment (sondreso) [reply .github/workflows/build.yml:29]: But we would still need to set the PYTHONPATH right?

Comment (sondreso) [reply .github/workflows/build.yml]: I will give it a go!

Comment (dotfloat) [reply .github/workflows/build.yml]: It *might* work, so why not. You might need to do `-DCMAKE_INSTALL_PREFIX=/usr`. But I don't think it's that important to warrant spending a lot of time on.

Comment (sondreso) [reply .github/workflows/build.yml]: `sudo make install` did not play nice with MacOS, so will leave it like it is for now :disappointed:

File: dev-requirements.txt [hunk 5]
```diff
@@ -1,9 +1,7 @@
-black==19.10b0 ; python_version >= '3.6'
+black==19.10b0
 pytest<6
 pytest-qt
-mock ; python_version >= '3'
```
Comment (ManInFez) [inline dev-requirements.txt:4]: should we not still keep mock?

Comment (dotfloat) [reply dev-requirements.txt:4]: https://github.com/equinor/ert/pull/1010#discussion_r485591209

We're using `unittest.mock` which is a part of CPython 3

**Review by jondequinor — APPROVED** [review]
LGTM

**Pull request merged**